meson.build

```diff
@@ -881,6 +881,18 @@ libm = cc.find_library('m')
 libdl = cc.find_library('dl')
 libcrypt = cc.find_library('crypt')
 
+crypt_header = conf.get('HAVE_CRYPT_H') == 1 ? \
+               '''#include <crypt.h>''' : '''#include <unistd.h>'''
+foreach ident : [
+        ['crypt_r',           crypt_header],
```

Comment:
crypt_r is glibc already, we should not need to test for it.

Reply:
Hah, now I see downgrading glibc on the current Arch Linux failed:

```
error: failed to commit transaction (conflicting files)
libxcrypt: /usr/include/crypt.h exists in filesystem (owned by glibc)
libxcrypt: /usr/lib/libcrypt.so exists in filesystem (owned by glibc)
``` 

The thing is that in recent glibc updates Arch transitioned from libcrypt in glibc to libcrypt in libxcrypt, see [[0]](https://github.com/archlinux/svntogit-packages/commit/6520f909c155e69c0fe6a494df3dcf071e8e4506#diff-91e904bd64036c9c236ce84542f97781R198)[[1]](https://github.com/archlinux/svntogit-packages/commit/53bcabefac14b9ce12c6273fa3961e9103cad8c5#diff-8d0411b338c83cd8cd8ad9d9db127101). And, ofc course, the libxcrypt contains the functions you seek:

```
libxcrypt /usr/share/man/man3/crypt_r.3.gz
libxcrypt /usr/share/man/man3/crypt_ra.3.gz
libxcrypt /usr/share/man/man3/crypt_rn.3.gz
```

But right now the upgrade is blocked by the sanitizer issue, but to make it go away we need to upgrade the image, so we're in a chicken-egg situation. I guess I could temporarily disable the test which triggers the issue (TEST-46) and go ahead with the upgrade.

Reply:
all i wanted to say: there's no point in checking for crypt_r, since our baseline, i.e. old glibc had that since time began.

checking for crypt_ra() definitely makes sense since glibc never had that, and onl libxcrypt has.

crypt_rn is a call we probably don#t care about, we shouldn't test for it hence